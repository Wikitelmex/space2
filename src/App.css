body {
  margin: 5px;
}

html {
  overflow-y:scroll;
}

ul {
  list-style-type: none;
}

.rocket-img {
  width: 100%;
}

.btn-reserve,
.btn-cancel {
  padding: 6px;
}

.btn-cancel {
  width: 200px;
}

.btn-reserve {
  width: 150px;
}
