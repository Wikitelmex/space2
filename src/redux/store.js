import { createStore, applyMiddleware, combineReducers } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

// import reducers here
import missionsReducer from './missions/missionReducer.js';

const mixReducer = combineReducers({
  // add reducers here
  missions: missionsReducer,
});

const store = createStore(
  mixReducer,
  applyMiddleware(logger, thunk),
);

export default store;
