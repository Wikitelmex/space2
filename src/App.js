import {
  BrowserRouter as Router,
  Switch,
  Route,
} from 'react-router-dom';
import './App.css';

import { Provider } from 'react-redux';
import store from './redux/store';

import Navbar from './Components/Navbar';
import ProfilePage from './Pages/ProfilePage';
import RocketsPage from './Pages/RocketsPage';
import MissionsPage from './Pages/MissionsPage';
// import FetchLoaderComponent from './Components/FetchLoaderComponent';

function App() {
  return (
    <Provider store={store}>
      <Router>
        <Navbar />
        <br />
        <br />
        <br />
        <div>
          <Switch>
            <Route exact path="/">
              <ProfilePage />
            </Route>
            <Route path="/rockets">
              <RocketsPage />
            </Route>
            <Route path="/missions">
              <MissionsPage />
            </Route>
          </Switch>
        </div>
      </Router>
    </Provider>
  );
}

export default App;
