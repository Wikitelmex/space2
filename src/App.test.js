import { render, screen, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../src/redux/store';
import App from './App';

afterEach(cleanup);

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByTestId("navBar");
  expect(linkElement).toBeInTheDocument();
});

it("renders App page with redux", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <App />
    </Provider>);
  expect(asFragment()).toMatchSnapshot();
});

