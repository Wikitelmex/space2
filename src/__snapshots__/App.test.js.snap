// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App page with redux 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-light bg-white fixed-top border-bottom shadow"
  >
    <div
      class="container-fluid ms-5"
    >
      <a
        aria-current="page"
        class="navbar-brand h1 active"
        href="/"
      >
        <h3
          class="fw-bold mt-1"
        >
          <img
            alt="Girl in a jacket"
            class="me-2"
            height="40"
            src="SpaceLogo.png"
            width="40"
          />
          Space Travelers' Hub
        </h3>
      </a>
      <button
        aria-controls="navbarNavAltMarkup"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarNavAltMarkup"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse mb-2 me-auto"
        id="navbarNavAltMarkup"
      >
        <ul
          class="navbar-nav ms-auto text-light me-3"
        >
          <li
            class="nav-link me-3"
          >
            <a
              aria-current="page"
              class="active-link"
              href="/"
            >
              Profile
            </a>
          </li>
          <li
            class="nav-link me-3"
          >
            <a
              href="/rockets"
            >
              Rockets
            </a>
          </li>
          <li
            class="nav-link me-3"
          >
            <a
              href="/missions"
            >
              Missions
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <br />
  <br />
  <br />
  <div
    data-testid="navBar"
  >
    <div>
      <h1>
        Profile Page
      </h1>
      <div>
        <h2>
          My Missions
        </h2>
        <div>
          No Missions loaded yet...
        </div>
      </div>
      <div>
        <h2>
          My Rockets
        </h2>
        <ul />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
